controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentOrder.java

```diff
@@ -113,35 +109,33 @@ public boolean isLast(JobType job, Application application) {
     public List<JobType> jobsFrom(DeploymentSpec deploymentSpec) {
         return deploymentSpec.steps().stream()
                 .flatMap(step -> jobsFrom(step).stream())
-                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
+                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
     }
 
     /** Returns job status sorted according to deployment spec */
-    public Map<JobType, JobStatus> sortBy(DeploymentSpec deploymentSpec, Map<JobType, JobStatus> jobStatus) {
-        List<DeploymentJobs.JobType> jobs = jobsFrom(deploymentSpec);
-        return jobStatus.entrySet().stream()
-                .sorted(Comparator.comparingInt(kv -> jobs.indexOf(kv.getKey())))
-                .collect(Collectors.collectingAndThen(toLinkedMap(Map.Entry::getKey, Map.Entry::getValue),
-                                                      Collections::unmodifiableMap));
+    public List<JobStatus> sortBy(DeploymentSpec deploymentSpec, Collection<JobStatus> jobStatus) {
+        List<DeploymentJobs.JobType> sortedJobs = jobsFrom(deploymentSpec);
+        return jobStatus.stream()
+                .sorted(comparingInt(job -> sortedJobs.indexOf(job.type())))
+                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
     }
 
     /** Returns deployments sorted according to declared zones */
-    public Map<Zone, Deployment> sortBy(List<DeploymentSpec.DeclaredZone> zones, Map<Zone, Deployment> deployments) {
+    public List<Deployment> sortBy(List<DeploymentSpec.DeclaredZone> zones, Collection<Deployment> deployments) {
         List<Zone> productionZones = zones.stream()
-                .filter(z -> z.environment() == Environment.prod && z.region().isPresent())
+                .filter(z -> z.region().isPresent())
                 .map(z -> new Zone(z.environment(), z.region().get()))
-                .collect(Collectors.toList());
-        return deployments.entrySet().stream()
-                .sorted(Comparator.comparingInt(kv -> productionZones.indexOf(kv.getKey())))
-                .collect(Collectors.collectingAndThen(toLinkedMap(Map.Entry::getKey, Map.Entry::getValue),
-                                                      Collections::unmodifiableMap));
+                .collect(toList());
+        return deployments.stream()
+                .sorted(comparingInt(deployment -> productionZones.indexOf(deployment.zone())))
+                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
     }
```

Comment:
Changed these two to accept collections and return lists, to clarify intent, and to avoid needless complexity.

Reply:
👍